src/cpp/session/modules/SessionRCompletions.R

```diff
@@ -117,6 +117,133 @@ assign(x = ".rs.acCompletionTypes",
       .rs.acCompletionTypes$UNKNOWN
 })
 
+.rs.addFunction("convertRoxygenDescription", function(description) {
+   file.md <- tempfile(fileext = ".md")
+   file.html <- tempfile(fileext = ".html")
+   on.exit({
+      unlink(file.md)
+      unlink(file.html)
+   })
+   writeLines(description, file.md)
+   rmarkdown::pandoc_convert(file.md, output = file.html)
+   paste(readLines(file.html), collapse = " ")
+})
+
+.rs.addJsonRpcHandler("convert_roxygen_description", function(description) {
+   .rs.scalar(.rs.convertRoxygenDescription(description))
+})
+
+.rs.addFunction("roxygenTagMetadata", function() {
+   if (.rs.isPackageVersionInstalled("roxygen2", "7.2.1"))
+   {
+      tagsFile <- system.file("roxygen2-tags.yml", package = "roxygen2")
+      contents <- readLines(tagsFile, warn = FALSE)
+      yaml <- .rs.fromYAML(contents)
+      n <- length(yaml)
+
+      map_chr <- function(...) vapply(..., FUN.VALUE = character(1), USE.NAMES = FALSE)
```

Comment:
It might be useful promoting these to `Tools.R` as e.g. `.rs.mapChr()` and friends.

Reply:
It could be interesting to somehow import the `compat-purrr.R` file, which we sometimes use when we want something like `purrr` without taking the dependency. 

https://github.com/tidyverse/dplyr/blob/main/R/compat-purrr.R